docs/Developers/Writing_bear_actions.rst

```diff
@@ -0,0 +1,174 @@
+
+.. code:: python
+
+    from coalib.results.result_actions.ResultAction import ResultAction
+
+    class SomeAction(ResultAction):
+
+        SUCCESS_MESSAGE = 'Something done successfully.'
+
+``ResultAction`` is used as a base class for all actions and provides a unified
+interface for all actions.
+
+``SUCCESS_MESSAGE`` is the message that will get printed once the action is
+successfully applied.
+
+Every action class should have these two methods:
+
+1. ``is_applicable`` method:
+   It has the logic which determines
+   whether the action is applicable or not.
+
+   .. code:: python
+
+           @staticmethod
+           def is_applicable(result,
+                             original_file_dict,
+                             file_diff_dict,
+                             applied_actions=()):
+
+   It must return a boolean value, ``True`` if action is applicable for a
+   particular result, ``False`` otherwise.
+
+   ``is_applicable`` method accepts four parameter:
+
+   -  ``result``: The result for which we want to check if action is appilcable
+      or not.
+   -  ``original_file_dict``: A dictionary whose key is the filename and value
+      is the file in the state when result was generated.
+   -  ``file_diff_dict``: A dictionary whose key is the filename and value
+      is a diff of file from the state in the original_file_dict to
+      the current state.
+   -  ``applied_actions``: List of actions names that have already been applied
+      for the current result.
+
+   .. note::
+
+       ``is_applicable`` doesn’t have to be a static method. In this case you
+       also need to prepend ``self`` to the parameters in the signature.
+
+2. ``apply`` method:
+   It method has the logic for when the action is applied.
+
+   .. code:: python
+
+           def apply(self,
+                     result,
+                     original_file_dict,
+                     file_diff_dict,
+                     **kwargs):
+               """
+               Apply (S)ome Action
+               """
+
+   apply method must return ``file_diff_dict``.
+
+   Writing a docstring for ``apply`` method is very important. The contents of
+   the docstring will get displayed to the user when asked to choose an action
+   to apply. The letter in parenthesis can be used by the user as a way of
+   selecting this action.
+
+Apart from this you can also add other helper methods and even an ``init``
+method to the action class.
+
+Passing bear action to the result
+---------------------------------
+
+After writing bear action you have to let coala know about this.
+When a bear yields a result you can pass a list action instances for that
+bear by using a optional argument ``actions``.
+
+Suppose ``SomeAction`` is an action specific to ``SomeBear``
+
+.. code:: python
+
+    from path.to.SomeAction import SomeAction
+
+    class SomeBear:
+
+        def run():
+
+            yield Result(.., actions=[SomeAction()])  # Ignore RSTcheckBear
```

Comment:
RSTcheckBear having some issues with this line?

Reply:
Yes, it is complaining about invalid syntax, it is because of `..`

Reply:
Maybe then you shouldn't ignore, complete the syntax.